src/wallet/wallet_rpc_server.cpp

```diff
@@ -2209,6 +2210,71 @@ namespace tools
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
+  bool wallet_rpc_server::on_encrypt(const wallet_rpc::COMMAND_RPC_ENCRYPT::request& req, wallet_rpc::COMMAND_RPC_ENCRYPT::response& res, epee::json_rpc::error& er, const connection_context *ctx)
+  {
+    if (!m_wallet) return not_open(er);
+    if (m_restricted)
+    {
+      er.code = WALLET_RPC_ERROR_CODE_DENIED;
+      er.message = "Command unavailable in restricted mode.";
+      return false;
+    }
+
+    try
+    {
+      const auto& secret_key = m_wallet->get_account().get_keys().m_spend_secret_key;
+      auto padded_plaintext = string_tools::pad_to_div_by(4, req.plaintext);
+      res.pad = padded_plaintext.size() - req.plaintext.size();
+      auto ciphertext = m_wallet->encrypt(padded_plaintext, secret_key, req.authenticated);
+      const std::vector<uint8_t> binary(begin(ciphertext), end(ciphertext));
+      std::vector<char> z85(binary.size()*5/4 + 1);
+      if (not zmq_z85_encode(z85.data(), binary.data(), binary.size())) {
+        throw std::logic_error("z85 encode unsuccessful");
```

Comment:
Does this cause an actual sane RPC error packet back (instead of returning false with an error string) ?

Reply:
If I change it to
```cpp
     if (zmq_z85_encode(z85.data(), binary.data(), binary.size())) {
        throw std::logic_error("z85 encode unsuccessful");
      }
```
it produces
```sh
Starting servers...
[TEST STARTED] encrypt_decrypt_message
Creating wallets
Encrypting/decrypting strings with spend key without authentication
Traceback (most recent call last):
  File "<monero>/tests/functional_tests/encrypt_decrypt_message.py", line 78, in <module>
    StringEncryptionTest().run_test()
  File "<monero>/tests/functional_tests/encrypt_decrypt_message.py", line 47, in run_test
    self.check_encrypt_decrypt(text, authenticated)
  File "<monero>/tests/functional_tests/encrypt_decrypt_message.py", line 71, in check_encrypt_decrypt
    encrypt_result = self.wallet[0].encrypt(plaintext, authenticated)
  File "<monero>/utils/python-rpc/framework/wallet.py", line 746, in encrypt
    return self.rpc.send_json_rpc_request(encrypt)
  File "<monero>/utils/python-rpc/framework/rpc.py", line 81, in send_json_rpc_request
    return self.send_request("/json_rpc", inputs, 'result')
  File "<monero>/utils/python-rpc/framework/rpc.py", line 73, in send_request
    assert 'error' not in res, res
AssertionError: {'error': {'code': -49, 'message': 'z85 encode unsuccessful'}, 'id': '0', 'jsonrpc': '2.0'}
[TEST FAILED] encrypt_decrypt_message
Stopping servers...
Done, 1/1 tests failed: encrypt_decrypt_message
```
Is this good or there is a better way?